src/bun.js/net.exports.js

```diff
@@ -53,9 +53,246 @@ export function isIP(s) {
   return 0;
 }
 
+const { Duplex } = import.meta.require("node:stream");
+
+export class Socket extends Duplex {
+  bytesRead = 0;
+  bytesWritten = 0;
+  connecting = false;
+  localAddress = "127.0.0.1";
+  #readQueue = [];
+  remotePort;
+  #socket;
+  timeout = 0;
+  #writeQueue;
+
+  constructor(options) {
+    super({
+      allowHalfOpen: options?.allowHalfOpen || false,
+      readable: true,
+      writable: true,
+    });
+    options?.signal?.once("abort", () => this.destroy());
+    this.once("connect", () => this.emit("ready"));
+    // TODO support `options.fd`
+  }
+
+  address() {
+    return {
+      address: this.localAddress,
+      family: this.localFamily,
+      port: this.localPort,
+    };
+  }
+
+  get bufferSize() {
+    return this.writableLength;
+  }
+
+  connect(port, host, connectListener) {
+    // TODO support IPC sockets
+    if (typeof host == "function") {
+      connectListener = host;
+      host = undefined;
+    }
+    if (typeof port == "object") {
+      var {
+        port,
+        host,
+        // TODOs
+        localAddress,
+        localPort,
+        family,
+        hints,
+        lookup,
+        noDelay,
+        keepAlive,
+        keepAliveInitialDelay,
+      } = port;
+    }
+    const self = this;
+    let ended = false;
+    self.connecting = true;
+    self.remotePort = port;
+    if (connectListener) self.on("connect", connectListener);
+    Bun.connect({
+      hostname: host || "localhost",
```

Comment:
localhost and not `0.0.0.0`, right?

Reply:
https://nodejs.org/dist/latest-v18.x/docs/api/net.html#socketconnectoptions-connectlistener
> For TCP connections, available options are:
>
>    port &lt;number> Required. Port the socket should connect to.
>    host &lt;string> Host the socket should connect to. **Default:** `'localhost'`.